src/perl/Permabit/VolumeGroup.pm

```diff
@@ -145,29 +146,24 @@ sub createLogicalVolume {
   # a yes/no on whether to wipeout a filesystem signature (it does on RHEL7),
   # the input redirection will cause the answer to be "no". The volume is not
   # immediately enabled.
-  $machine->runSystemCmd("sudo lvcreate --name $name --size ${ksize}K --yes"
+  $machine->runSystemCmd("sudo lvcreate --name $name -an -kn --size ${ksize}K --yes"
                          . " $config $self->{volumeGroup} </dev/null");
-  # Make sure to wait for the udev event recording the new event has
-  # been processed. Otherwise, the lv may be open due to blkid when
-  # we try to disable it below.
-  $machine->sendCommand("sudo udevadm settle");
-  # XXX As soon as we get rid of Squeeze, on which lvcreate doesn't have the
-  # -a flag, we can stop doing two separate commands and use lvcreate -an.
-  $self->disableLogicalVolume($name);
 }
 
 ########################################################################
-# Create a thin volume
+# Create a thin pool
 #
-# @param name  Thin volume name
-# @param size  Thin volume size in bytes
+# @param  name    Thin pool name
+# @param  size    Thin pool size in bytes
+# @oparam config  String of create time lvm.conf overrides
+# @oparam vdo     Whether the thin pool uses VDO or not
 ##
-sub createThinVolume {
-  my ($self, $name, $size) = assertNumArgs(3, @_);
+sub createThinPool {
+  my ($self, $name, $size, $config, $vdo) = assertMinMaxArgs(["", 0], 3, 5, @_);
```

Comment:
$vdo here is a boolean, yes? Could you name it some thing like "useVdo" so it's clear that it's a boolean, and what the flag means? Just calling it $vdo makes me expect a reference to an object representing a VDO device or something.

Reply:
renamed